WalletWasabi.Gui/Controls/QrCode.cs

```diff
@@ -25,46 +48,74 @@ static QrCode()
 			set => SetAndRaise(MatrixProperty, ref _matrix, value);
 		}
 
+		private bool[,] AddPaddingToMatrix(bool[,] matrix)
+		{
+			var dims = GetMatrixDimensions(matrix);
+			var nW = dims.W + (_matrixPadding * 2);
+			var nH = dims.H + (_matrixPadding * 2);
+
+			var paddedMatrix = new bool[nH, nW];
+
+			for (var i = 0; i < dims.H; i++)
+			{
+				for (var j = 0; j < dims.W; j++)
+				{
+					paddedMatrix[i + _matrixPadding, j + _matrixPadding] = matrix[i, j];
+				}
+			}
+
+			return paddedMatrix;
+		}
+
 		public override void Render(DrawingContext context)
 		{
-			var source = Matrix;
+			var source = FinalMatrix;
 
-			if (source != null)
+			if (source is null)
 			{
-				var h = source.GetUpperBound(0) + 1;
-				var w = source.GetUpperBound(1) + 1;
+				return;
+			}
 
-				var minDimension = Math.Min(w, h);
-				var minBound = Math.Min(Bounds.Width, Bounds.Height);
-				var factor = (float)minBound / minDimension;
+			var dims = GetMatrixDimensions(source);
 
-				for (var i = 0; i < h; i++)
+			context.FillRectangle(Brushes.White, new Rect(0, 0, GridCellFactor * dims.W, GridCellFactor * dims.H));
+
+			for (var i = 0; i < dims.H; i++)
+			{
+				for (var j = 0; j < dims.W; j++)
 				{
-					for (var j = 0; j < w; j++)
-					{
-						var cellValue = source[i, j];
-						var rect = new Rect(i * factor, j * factor, factor, factor);
-						var color = cellValue ? Brushes.Black : Brushes.White;
-						context.FillRectangle(color, rect);
-					}
+					var cellValue = source[i, j];
+					var rect = new Rect(i * GridCellFactor, j * GridCellFactor, GridCellFactor, GridCellFactor);
+					var color = cellValue ? Brushes.Black : Brushes.White;
+					context.FillRectangle(color, rect);
 				}
 			}
 		}
 
+		private (int W, int H) GetMatrixDimensions(bool[,] source)
```

Comment:
Since `w` and `h` are fields they should be `lowerCase`.
Also use block body for methods please.